services/pull/lfs.go

```diff
@@ -101,10 +103,16 @@ func createLFSMetaObjectsFromCatFileBatch(catFileBatchReader *io.PipeReader, wg
 		}
 		pointerBuf = pointerBuf[:size]
 		// Now we need to check if the pointerBuf is an LFS pointer
-		pointer := lfs.IsPointerFile(&pointerBuf)
-		if pointer == nil {
+		pointer, err := lfs.ReadPointerFromBuffer(pointerBuf)
+		if err != nil {
```

Comment:
Add an error log here is better.

Reply:
But it is not an error in this case if a file is not a LFS pointer or it doesn't exist in the store.

Reply:
Is there a way to check that it is not a legitimate error? That is, if it is a pointer file, but cannot resolve it in the LFS store it should throw an error, right? 

Otherwise, maybe write to info/debug log and a comment saying "File not found in LFS store
 Continuing."

Reply:
You are permitted to create files with pointer like content in your repository. All of these files will trigger that logging point because their content will not be in the store. An other example would be the use of a different LFS endpoint than Gitea.

Reply:
I think the problem which may lead to confusion is that we are using `err` as a valid code path. A better solution would be to test the condition you mentioned, then catch any other error. 
```
if not LFS pointer: continue
if LFS pointer not in store: warn & continue
else: read & catch error
```

Reply:
see https://github.com/go-gitea/gitea/pull/14726#pullrequestreview-596715959
The error is not a real error here but an "not a pointer" indicator. Maybe I could add a `Pointer.IsValid()` method which checks if oid and size are set.

Reply:
> Maybe I could add a `Pointer.IsValid()` method

Yeah, that's a good solution I think. Imo the point is `err` should not be used as valid code path. Even `if pointer == nil` is more clear.

@lunny does this resolve your concern?

Reply:
@kdumontnu  Both OK. And if we use @KN4CK3R 's solution. `IsValid` is necessary and we have to check the pointer when requested as a http param.

Reply:
I have added `IsValid()` with some unit tests.